esx_service/vmdk_ops.py

```diff
@@ -314,15 +314,25 @@ def find_child(vm_name):
 
 #returns error, or None for OK
 def attachVMDK(vmdk_path, vm_name):
-    vm = findVmByName(vm_name)
+    try:
+       vm = findVmByName(vm_name)
+    except Exception as e:
+       msg = "VM {0} not found".format(vm_name)
+       return err(msg)
+
     logging.info("*** attachVMDK: %s to %s VM uuid = %s",
                  vmdk_path, vm_name, vm.config.uuid)
     return disk_attach(vmdk_path, vm)
 
 
 #returns error, or None for OK
 def detachVMDK(vmdk_path, vm_name):
-    vm = findVmByName(vm_name)
+    try:
+       vm = findVmByName(vm_name)
+    except Exception as e:
```

Comment:
I agree with Prashant - it looks strange to mix-n-match exceptions and return code without any visible system
We can agree on "catch exceptoins as soon as they happen and pass error code around" or "catch exceptions only when decision needs to be made". findVmByName handling mixes both - I think it needs to be changed to a single style